radix-engine/src/system/system_modules/node_move/node_move_module.rs

```diff
@@ -99,39 +104,44 @@ impl NodeMoveModule {
                 }
 
                 // Change to restricted unless it's moved to auth zone.
-                // TODO: align with barrier design?
-                let mut changed_to_restricted = true;
-                if let Some(method) = callee.try_as_method() {
-                    let type_info = TypeInfoBlueprint::get_type(&method.node_id, api)?;
-                    if let TypeInfoSubstate::Object(ObjectInfo { blueprint, .. }) = type_info {
-                        if blueprint.eq(&Blueprint::new(&RESOURCE_PACKAGE, AUTH_ZONE_BLUEPRINT)) {
-                            changed_to_restricted = false;
-                        }
+                if callee.is_barrier() {
+                    let handle = api.kernel_lock_substate(
+                        &node_id,
+                        OBJECT_BASE_PARTITION,
+                        &FungibleProofField::Moveable.into(),
```

Comment:
This should technically be `NonFungibleProofField::Moveable`, and same on line 131.

Reply:
Good catch!